src/main/java/control/FBase.java

```diff
@@ -53,9 +53,11 @@ public FBase(String configName) {
 	public void startup(boolean registerAtNamingService) throws InterruptedException,
 			ExecutionException, TimeoutException, FBaseStorageConnectorException {
 		if (Connector.S3.equals(configuration.getDatabaseConnector())) {
```

Comment:
We could replace this entire if/else block with
connector = (AbstractDBConnector) Class.forName(classNameFromSomeConfig).newInstance();